akka-http/src/main/scala/akka/http/scaladsl/server/directives/HeaderDirectives.scala

```diff
@@ -171,8 +171,12 @@ object HeaderMagnet extends LowPriorityHeaderMagnetImplicits {
 
   /**
    * If possible we want to apply the special logic for [[ModeledCustomHeader]] to extract custom headers by type,
-   * otherwise the default `fromUnit` is good enough (for headers that the parser emits in the right type already).
+   * otherwise the default `fromCompanion` is good enough (for headers that the parser emits in the right type already).
    */
+  implicit def fromCompanionForModeledCustomHeader[T <: ModeledCustomHeader[T], H <: ModeledCustomHeaderCompanion[T]](companion: ModeledCustomHeaderCompanion[T])(implicit tag: ClassTag[T]): HeaderMagnet[T] =
+    fromClassTagForModeledCustomHeader[T, H](tag, companion)
+
+  @deprecated("Pass the companion object to headerValueByType as a parameter instead", since = "10.2.0")
```

Comment:
Let's add an example to that particular deprecation note to make it concrete how to do the conversion as so many usages will be affected.